main/src/org/destinationsol/IniReader.java

```diff
@@ -19,74 +19,95 @@
 import com.badlogic.gdx.files.FileHandle;
 import org.destinationsol.files.FileManager;
 
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.util.ArrayList;
 import java.util.HashMap;
 import java.util.List;
 
 public class IniReader {
 
-  private final HashMap<String,String> myVals;
-
-  public IniReader(String fileName, SolFileReader reader, boolean readOnly) {
-    myVals = new HashMap<String, String>();
-    List<String> lines = reader != null ? reader.read(fileName) : fileToLines(fileName, readOnly);
-
-    for (String line : lines) {
-      int commentStart = line.indexOf('#');
-      if (commentStart >= 0) {
-        line = line.substring(0, commentStart);
-      }
-      String[] sides = line.split("=");
-      if (sides.length < 2) continue;
-      String key = sides[0].trim();
-      String val = sides[1].trim();
-      myVals.put(key, val);
+    private final HashMap<String, String> myVals;
+
+    public IniReader(String fileName, SolFileReader reader, boolean readOnly) {
+        myVals = new HashMap<>();
+        List<String> lines = reader != null ? reader.read(fileName) : fileToLines(fileName, readOnly);
+        initValueMap(lines);
     }
-  }
 
-  private List<String> fileToLines(String fileName, boolean readOnly) {
-    FileManager.FileLocation accessType = readOnly ? FileManager.FileLocation.STATIC_FILES : FileManager.FileLocation.DYNAMIC_FILES;
-    FileHandle fh = FileManager.getInstance().getFile(fileName, accessType);
+    public IniReader(BufferedReader reader) {
```

Comment:
This `IniReader` constructor accepts a `BufferedReader` stream instead of a file path - this is particularly useful for reading "ini" text files directly from streams declared in `FileFormats`, avoiding path reconstruction hacks.